tensorflow/compiler/jit/cluster_scoping_pass_test.cc

```diff
@@ -0,0 +1,172 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/compiler/jit/cluster_scoping_pass.h"
+
+#include "tensorflow/compiler/jit/defs.h"
+#include "tensorflow/core/framework/node_def_util.h"
+#include "tensorflow/core/framework/op.h"
+#include "tensorflow/core/graph/algorithm.h"
+#include "tensorflow/core/graph/graph_constructor.h"
+#include "tensorflow/core/graph/graph_def_builder.h"
+#include "tensorflow/core/graph/graph_def_builder_util.h"
+#include "tensorflow/core/lib/core/status_test_util.h"
+#include "tensorflow/core/platform/test.h"
+#include "tensorflow/core/public/session_options.h"
+
+namespace tensorflow {
+namespace {
+
+Status ClusterScoping(std::unique_ptr<Graph>* graph) {
+  FixupSourceAndSinkEdges(graph->get());
+
+  GraphOptimizationPassOptions opt_options;
+  opt_options.graph = graph;
+  FunctionDefLibrary fdef_lib;
+  FunctionLibraryDefinition flib_def(OpRegistry::Global(), fdef_lib);
+  opt_options.flib_def = &flib_def;
+  SessionOptions session_options;
+  session_options.env = Env::Default();
+  session_options.config.mutable_graph_options()
+      ->mutable_optimizer_options()
+      ->set_global_jit_level(OptimizerOptions::ON_2);
+  opt_options.session_options = &session_options;
+
+  ClusterScopingPass pass;
+  return pass.Run(opt_options);
+}
+
+std::unordered_map<string, string> GetXlaScopes(const Graph& graph) {
```

Comment:
Use `absl::flat_hash_map`.

Usually in new code we prefer using ABSL versions whenever possible.  There is some old code that uses std versions of maps and sets because originally TF did not include ABSL.